lib/server/middleware/maintenance.js

```diff
@@ -0,0 +1,51 @@
+import wildcard from 'node-wildcard';
+import boolean from 'boolean';
+import SettingModel from '../models/setting';
+import PageModel from '../models/page';
+import parseSettings from 'helpers/parse-settings';
+
+const ignoreMaintenanceRedirect = (url, options) => {
+  let result = false;
+  let i = 0;
+  const patterns = ['/admin*', '/images*', '/fonts*', '/graphql*', '/assets*', options.maintenancePage];
+  for (i = 0; i < patterns.length; i++) {
+    result = wildcard(url, patterns[i]);
+    if (result) {
+      break;
+    }
+  }
+  return result;
+};
+
+export default async (req, res, next) => {
+  const settingsArr = await SettingModel
+    .find({
+      _id: {$in: ['maintenanceMode', 'useDefaultMaintenancePage', 'maintenancePage']}
+    })
+    .exec();
+  const settings = parseSettings(settingsArr);
+  const defaultMaintenancePage = '/pub-res/maintenance';
+  let maintenancePage = defaultMaintenancePage;
+  settings.useDefaultMaintenancePage = boolean(settings.useDefaultMaintenancePage);
+  settings.maintenanceMode = boolean(settings.maintenanceMode);
+
+  if (settings.maintenancePage && !settings.useDefaultMaintenancePage) {
+    const page = await PageModel
+      .findById(settings.maintenancePage)
+      .exec();
+    maintenancePage = (!page) ? maintenancePage : `/${page.slug}`;
+  }
+
+  if (!settings.maintenanceMode &&
+    (req.url.toLowerCase() === maintenancePage || req.url.toLowerCase() === defaultMaintenancePage)) {
+    return res.redirect('/');
+  }
+
+  if (ignoreMaintenanceRedirect(req.url, {maintenancePage}) ||
+      req.user ||
+      !settings.maintenanceMode) {
+    return next();
+  }
+
+  res.redirect(maintenancePage);
+};
```

Comment:
for this we would just send up raw html instead of creating a react component?

This may also mean creating a custom maintenance page will be manual as well and be configured from the backend.